incubator/hnc/third_party/open-policy-agent/gatekeeper/pkg/webhook/certs.go

```diff
@@ -503,20 +538,50 @@ func (r *ReconcileVWH) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return reconcile.Result{Requeue: true}, err
 	}
 
+	crds := []*unstructured.Unstructured{}
+	crd := &unstructured.Unstructured{}
+	crd.SetGroupVersionKind(crdGVK)
```

Comment:
I think this will fail as soon as there's more than one CRD type - on line 555, you add this _pointer_ to an array, but you keep reading into the same pointer. So if there are three CRDs, I _think_ that `crds` will land up with three copies of whatever the last CRD was.

As a general rule, I like to define my variables at the lowest scope possible, which in this case is inside the loop, not outside.

Reply:
I tested this pattern in the go playground with a fake struct to make sure it worked, but anyway I cannot remember now why I preferred putting this block outside the loop then. Agreed that it's better to place it inside the loop.